addons/mail/static/src/models/messaging_menu/messaging_menu.js

```diff
@@ -31,40 +31,31 @@ function factory(dependencies) {
          */
         toggleOpen() {
             this.update({ isOpen: !this.isOpen });
-            this.env.messaging.refreshIsNotificationPermissionDefault();
+            this.messaging.refreshIsNotificationPermissionDefault();
+            if (this.isOpen) {
+                // populate some needaction messages on threads.
+                this.env.messaging.inbox.cache.update({ isCacheRefreshRequested: true });
+            }
```

Comment:
I'm wondering what's the motivation of changes from `env.messaging` to `messaging` in this PR?
Is it because `env.messaging` doesn't mark field access properly? Or just 4 less characters to type/read?

If it's the former, it sucks a bit, but you have to do it on all of them, which is not the case (here for instance).
If it's the latter, I much prefer using `env.messaging` everywhere over `.messaging`.
This is for consistency, and also because I prefer global access rather than field access.

Note that `messaging` and other system singletons are arbitrary code splitting. There is plan to just dump everything in a single singleton record ("env") and split concerns on this object via some means to define features.

Reply:
Indeed `env.messaging` is not counted as an access, but that's only an issue when `env.messaging` is used by itself.
If any field of `env.messaging` is accessed, it will count as an access. So it's really only an issue on code where `!env.messaging` is a preventive return. By the way this is also why I needed to add `useMessaging` hook (now part of `useModels`).

Independently of this issue, I think it's a mistake to have a global access inside our own code, and I would much prefer if messaging was always forwarded as a relation to the models that need it. I also think we should remove `all()`, `create()`, `insert()` and pretty much all other static model methods, and only allow the corresponding action through writing on relational fields. `messaging` should be the only exception where a create is done outside of a relation, and it should be part of the bootstrapping code.

But here in particular I agree it makes no sense to change one occurrence in `toggleOpen` and not the other one. I meant to change it only inside the compute method but apparently I did other occurrences in the file. I guess I'll change the remaining one too.

Reply:
> Independently of this issue, I think it's a mistake to have a global access inside our own code, and I would much prefer if messaging was always forwarded as a relation to the models that need it. I also think we should remove `all()`, `create()`, `insert()` and pretty much all other static model methods, and only allow the corresponding action through writing on relational fields. `messaging` should be the only exception where a create is done outside of a relation, and it should be part of the bootstrapping code.

Indeed _everything_ should be models and fields. Static methods were kind-of mistakes, but at the time we introduced them we were not enlightened by turning everything to models and fields, and we were more at ease with implementation with OOP (static methods) and OWL features (env).

What I don't like with using `messaging` from field right now is that the dev has to write too many LOCs to define this field, and it's weird to define computed field for this. Should be bootstrapping code, so that we don't have this drawback.

But I completely agree with you.